components/shared/profile/edit/EditProfile.tsx

```diff
@@ -0,0 +1,451 @@
+          className="bg-black-700 text-white-100"
+          placeholder="portfolio"
+        />
+      </div>
+      {goalFields.map((field, index: number) => (
+        <>
+          <div key={field.id} className="profile-goals-wrapper">
+            <Input
+              type="checkbox"
+              color="green-500"
+              className="order-1 size-9 bg-black-700 text-white-100"
+              id={`learningGoals[${index}]CB`}
+              {...register(`learningGoals[${index}].completed` as any)}
+            />
+            <Input
+              className="order-2"
+              id={`learningGoals[${index}]`}
+              placeholder="Enter your goal"
+              {...register(`learningGoals[${index}].name` as any)}
+            />
+            <Button
+              type="button"
+              className="text-white-100"
+              onClick={() => removeGoal(index)}
+            >
+              Remove
+            </Button>
+          </div>
+        </>
+      ))}
+      <CustomButton
+        buttonType="profileButton"
+        type="button"
+        // @ts-ignore
+        onClick={() => appendGoal({ name: "", completed: false })}
+      >
+        Add Goal
+      </CustomButton>
+
+      {experienceFields.map((field, index) => {
+        return (
+          <>
+            <div className="profile-goals-wrapper" key={field.id}>
+              <Button
+                className="text-white-100"
+                type="button"
+                onClick={() => removeExperience(index)}
+              >
+                {" "}
+                Remove
+              </Button>
+              <Input
+                id={`experiences[${index}].name`}
+                type="text"
+                {...register(`experiences[${index}].name` as any)}
+              />
+            </div>
+          </>
+        );
+      })}
+      <CustomButton
+        buttonType="profileButton"
+        type="button"
+        onClick={() => appendExperience({ name: "" })}
+      >
+        Add Experience
+      </CustomButton>
+      {/* Technologies Section with Search Box */}
+      <div className="flex flex-col justify-stretch gap-2 px-3.5 py-3  ">
+        <label
+          htmlFor="technologies"
+          className="justify-start space-y-2 text-white-300"
+        >
+          Tech Stacks
+        </label>
+        <div className="flex w-full  bg-black-700 ">
+          {technologies && // eslint-disable-next-line array-callback-return
+            technologies?.map((tech: any, index) => {
+              const icon = techStackBadges.find((badge) => badge.name === tech);
+              if (icon)
+                return (
+                  <>
+                    <span
+                      key={index}
+                      className="tech-badges paragraph-3-medium shadow-custom w-1/2   flex-row justify-between bg-black-600 text-white-100"
+                    >
+                      {icon.icon({ size: 16 })}
```

Comment:
https://github.com/agordon123/git_note/assets/4108484/c7e3350a-2de5-4158-932b-5f10e458a3dd

What happened to the constants.tsx we worked on? Now it's back to constants.ts and doesn't have the components themselves. I thought we had a whole meeting where we turned constants.ts into TSX and did this:


```js
export const techStackBadges: TechStackBadges[] = [
  {
    name: "react",
    icon: <DiReact/>,
    color: "name-blue-500",
  },
  {
    name: "node",
    icon: <DiNodejs />,
    color: "bg-green-500",
  },
  {
    name: "html",
    icon: <DiHtml5 />,
    color: "bg-red-500",
  },
  {
    name: "css",
    icon: <DiCss3 />,
  },
  {
    name: "javascript",
    icon: <DiJavascript />,
  },
  {
    name: "python",
    icon: <DiPython />,
  },
  {
    name: "java",
    icon: <DiJava />,
  },
  {
    name: "github",
    icon: <DiGithub />,
  },
  {
    name: "tailwind",
    icon: <BiLogoTailwindCss />,
  },
  {
    name: "nextjs",
    icon: <TbBrandNextjs />,
  },
  {
    name: "laravel",
    icon: <DiLaravel />,
    color: "bg-red-500",
  },
  {
    name: "php",
    icon: <FaPhp />,
  },
  {
    name: "typescript",
    icon: <BiLogoTypescript />,
  },
];
```


And changed our icon rendering to look like this:

```js
    {icon?.icon}
```

Reply:
If you want size 16 you can add it in the constants.tsx

Reply:
This is what the style looks like to me. Can you screenshot what it looks like for you?

![image](https://github.com/agordon123/git_note/assets/4108484/d7d6c471-0c73-42af-82b8-d794e757a9c5)

Reply:
thats what it looks like to me, but yeah youre right we did have that convo and did do that, possibly didnt save it .  ill clean it up